CONTRIBUTING.md

```diff
@@ -0,0 +1,56 @@
+# Contributing to Roslyn
+
+Guidelines for contributing to the Roslyn repo.
+
+## Submitting Pull Requests
+
+- **DO** ensure submissions pass all Jenkins legs and are merge conflict free.
+- **DO** ensure submissions meet functional and performance expectations, including scenarios for which the team doesn’t yet have open source tests. This means you may be asked to fix and resubmit your pull request against a new open test case if it fails one of these tests.
+- **DO** follow the [.editorconfig](http://editorconfig.org/) settings for each directory. 
+- **DO NOT** submit changes beyond the level of a bug fix without discussing with the team first, or they will likely be declined.
+- **DO NOT** submit unsolicited language features.  All language feature discussion should begin on the [C# language](https://github.com/dotnet/csharplang#discussion) / [VB language](https://github.com/dotnet/vblang) repos.  Once a feature is championed and validated by LDM a developer will be assigned to help begin a prototype on this repo inside a feature branch.
+- **DO NOT** submit large code formatting changes. Such changes will be declined.
```

Comment:
Given we ourselves in the IDE do this on occasion, this isn't a correct statement. Generally I think the answer is "talk to us in advance and make a separate PR".